openshift-kube-apiserver/filters/deprecatedapirequest_filter.go

```diff
@@ -0,0 +1,52 @@
+package filters
+
+import (
+	"net/http"
+	"os"
+	"sync"
+
+	"k8s.io/apiserver/pkg/endpoints/request"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/openshift-kube-apiserver/filters/deprecatedapirequest"
+)
+
+var once sync.Once
+var controller deprecatedapirequest.Controller
+
+// WithDeprecatedApiRequestLogging adds an http handler that logs requests to deprecated apis.
+func WithDeprecatedApiRequestLogging(handler http.Handler, genericConfig *genericapiserver.Config) http.Handler {
+	once.Do(func() {
+		nodeName := os.Getenv("HOST_IP")
+		if hostname, err := os.Hostname(); err != nil {
+			nodeName = hostname
+		}
+		config := rest.CopyConfig(genericConfig.LoopbackClientConfig)
+		config.AcceptContentTypes = "application/json"
+		config.ContentType = "application/json"
+		controller = deprecatedapirequest.NewController(nodeName, config)
+		genericConfig.AddPostStartHookOrDie("deprecated-api-requests-filter", func(pshCtx genericapiserver.PostStartHookContext) error {
```

Comment:
don't mutate the passed genericConfig.  Pass the controller through the method instead and register the post starthook with other registrations.